### ISSUE TITLE
Terraform fails to create Security Group

### ISSUE THREAD

**Issue by aimannajjar** [localstack/localstack#1873]
Hello - It seems that we may have come across a bug in VPC security group implementation. Please see details below. 

### Discussion
Terraform AWS provider expects newly created VPC Security Groups to have a default ALLOW_ALL Egress rule (see [their logic here](https://github.com/terraform-providers/terraform-provider-aws/blob/98b8b848ca94031b20c3e626c9d40484e3af80de/aws/resource_aws_security_group.go#L281-L336)). However, it looks that Localstack does not create this rule by default. Therefore causing Terraform to panic when the rule is not found.

### Steps to Reproduce
1. Try to apply following Terraform code:
```
resource "aws_security_group" "eks-node" {
  name        = "some-sg"
  vpc_id      = var.vpc_id

  egress {
    from_port   = 0
    to_port     = 0
    protocol    = "-1"
    cidr_blocks = ["0.0.0.0/0"]
  }
}
```
2. Terraform will fail with this error:
```
Error: Error revoking default IPv6 egress rule for Security Group (sg-9f612228): SerializationError: failed to unmarshal error message
	status code: 400, request id: 
caused by: UnmarshalError: failed to unmarshal error message
	00000000  3c 3f 78 6d 6c 20 76 65  72 73 69 6f 6e 3d 22 31  |<?xml version="1|
00000010  2e 30 22 20 65 6e 63 6f  64 69 6e 67 3d 22 55 54  |.0" encoding="UT|
00000020  46 2d 38 22 3f 3e 0a 20  20 3c 45 72 72 6f 72 52  |F-8"?>.  <ErrorR|
00000030  65 73 70 6f 6e 73 65 3e  0a 20 20 20 20 3c 45 72  |esponse>.    <Er|
00000040  72 6f 72 73 3e 0a 20 20  20 20 20 20 3c 45 72 72  |rors>.      <Err|
00000050  6f 72 3e 0a 20 20 20 20  20 20 20 20 3c 43 6f 64  |or>.        <Cod|
00000060  65 3e 49 6e 76 61 6c 69  64 50 65 72 6d 69 73 73  |e>InvalidPermiss|
00000070  69 6f 6e 2e 4e 6f 74 46  6f 75 6e 64 3c 2f 43 6f  |ion.NotFound</Co|
00000080  64 65 3e 0a 20 20 20 20  20 20 20 20 3c 4d 65 73  |de>.        <Mes|
00000090  73 61 67 65 3e 54 68 65  20 73 70 65 63 69 66 69  |sage>The specifi|
000000a0  65 64 20 72 75 6c 65 20  64 6f 65 73 20 6e 6f 74  |ed rule does not|
000000b0  20 65 78 69 73 74 20 69  6e 20 74 68 69 73 20 73  | exist in this s|
000000c0  65 63 75 72 69 74 79 20  67 72 6f 75 70 3c 2f 4d  |ecurity group</M|
000000d0  65 73 73 61 67 65 3e 0a  20 20 20 20 20 20 20 20  |essage>.        |
000000e0  0a 20 20 20 20 20 20 3c  2f 45 72 72 6f 72 3e 0a  |.      </Error>.|
000000f0  20 20 20 20 3c 2f 45 72  72 6f 72 73 3e 0a 20 20  |    </Errors>.  |
00000100  3c 52 65 71 75 65 73 74  49 44 3e 37 61 36 32 63  |<RequestID>7a62c|
00000110  34 39 66 2d 33 34 37 65  2d 34 66 63 34 2d 39 33  |49f-347e-4fc4-93|
00000120  33 31 2d 36 65 38 65 45  58 41 4d 50 4c 45 3c 2f  |31-6e8eEXAMPLE</|
00000130  52 65 71 75 65 73 74 49  44 3e 0a 3c 2f 45 72 72  |RequestID>.</Err|
00000140  6f 72 52 65 73 70 6f 6e  73 65 3e                 |orResponse>|

caused by: expected element type <Response> but have <ErrorResponse>
```

**Comment by aimannajjar** [comment]
After digging further, I realized that Localstack relies on Moto for this functionality. It seems the issue has been fixed in Moto master branch: https://github.com/spulec/moto/issues/2551

**Comment by aimannajjar** [comment]
I take back my previous comment, it seems this is a separate issue. I have tested against Moto master branch and the issue persisted. I submitted a new issue at Moto's Github repo: https://github.com/spulec/moto/issues/2646

**Comment by whummer** [comment]
Thanks for the detailed report @aimannajjar , and thanks for providing the fix in https://github.com/spulec/moto/pull/2647 . Going to close this issue - the fix will be released once the PR is merged in moto and we upgrade to the latest version. Thanks!
